src/attributes.cpp

```diff
@@ -1590,44 +1590,65 @@ namespace attributes {
 
         int templateCount = 0;
         int parenCount = 0;
-        bool insideQuotes = false;
+        bool endOfArg = false;
         std::string currentArg;
         std::vector<std::string> args;
-        char prevChar = 0;
-        for (std::string::const_iterator
-                            it = argText.begin(); it != argText.end(); ++it) {
-            char ch = *it;
+        char quote = 0;
+        bool escaped = false;
+        typedef std::string::const_iterator it_t;
+        for (it_t it = argText.begin(); it != argText.end(); ++it) {
 
-            if (ch == '"' && prevChar != '\\') {
-                insideQuotes = !insideQuotes;
-            }
+            endOfArg = false;
+
+            // Store current character
+            char ch = *it;
 
-            if ((ch == ',') &&
+            // Ignore quoted strings and character values in single quotes
+            if ( ! quote && (ch == '"' || ch == '\''))
+                quote = ch;
+            else if (quote && ch == quote && ! escaped)
+                quote = 0;
+
+            // Escaped character inside quotes
+            if (escaped)
+                escaped = false;
+            else if (quote && ch == '\\')
+                escaped = true;
+
+            // Detect end of argument declaration
+            if ( ! quote &&
+                (ch == ',') &&
                 (templateCount == 0) &&
-                (parenCount == 0) &&
-                !insideQuotes) {
+                (parenCount == 0)) {
                 args.push_back(currentArg);
                 currentArg.clear();
-                continue;
-            } else {
-                currentArg.push_back(ch);
-                switch(ch) {
-                    case '<':
-                        templateCount++;
-                        break;
-                    case '>':
-                        templateCount--;
-                        break;
-                    case '(':
-                        parenCount++;			// #nocov
-                        break;				// #nocov
-                    case ')':
-                        parenCount--;			// #nocov
-                        break;				// #nocov
+                endOfArg = true;
```

Comment:
This looks correct to me but IMHO the `endOfArg` flag is a bit more confusing than just using `continue` to bypass  adding characters to the current argument. Is there a reason why you made that switch (perhaps I'm missing a nuance)? Or just style?

Reply:
Hi, I used it when the escape character was still handled with a `char prevChar` variable that was set at the end of the loop. So for consistency, I preferred to replace the `continue` with `endOfArg` in order for `prevChar` to be set even when a comma was met (it was not the case with the `continue` statement, `prevChar` was not updated).
Since I've now changed `prevChar` with the boolean variable `escaped` and `escaped` is set before testing for a comma that marks the end of an argument, we can go back to the `continue` statement. I will make the change.

Reply:
Thanks for pointing this you, by the way.